### PR TITLE
fix(python): Remove `is_numeric` check on `Series.std/var`

### PR DESCRIPTION
**Pull request by ion-elgreco**
Small follow-up fix for this PR: https://github.com/pola-rs/polars/pull/12865

### CODE REVIEW

File: py-polars/polars/series/series.py [hunk 1]
```diff
@@ -1784,8 +1784,6 @@ def std(self, ddof: int = 1) -> float | None:
         1.0
 
         """
-        if not self.dtype.is_numeric():
-            return None
```
[the remark this answers is not in the document]
Comment (stinodego) [reply py-polars/polars/series/series.py:1788]: Hm, we really need a generic typed Series 🤔 for now I guess update the return type to `float | timedelta` ?

File: py-polars/tests/unit/series/test_series.py [hunk 2]
```diff
@@ -2801,3 +2801,14 @@ def test_series_cmp_fast_paths() -> None:
         pl.Series([None], dtype=pl.Boolean)
         == pl.Series([False, False], dtype=pl.Boolean)
     ).to_list() == [None, None]
+
+
+def test_series_duration_std_var() -> None:
+    s = pl.Series([timedelta(days=1), timedelta(days=2), timedelta(days=4)])
+    assert s.std() == timedelta(days=1, seconds=45578, microseconds=180014)
+    assert s.var() is None  # Way to large value so overflows
```
[the remark this answers is not in the document]
Comment (stinodego) [reply py-polars/tests/unit/series/test_series.py:2809]: Still, if I call `var`, and Polars cannot give me the `var`, I expect an error. We cannot return the `var` for some inputs and `None` for others.

[the remark this answers is not in the document]
Comment (ion-elgreco) [reply py-polars/tests/unit/series/test_series.py:2809]: @stinodego we could also return var always as float? But the number will always be represented in milli/micro/nano-seconds, so we should probably make a note of that

**Comment by ion-elgreco** [comment]
@stinodego I am addressing the overflow issue by casting to ms first in my other PR. If we merge that other one first, then I can update the test here, to reflect that change.

**Comment by stinodego** [comment]
> @stinodego I am addressing the overflow issue by casting to ms first in my other PR. If we merge that other one first, then I can update the test here, to reflect that change.

I don't think that's a valid way of doing things. You would lose a lot of precision on values tha require ns/us precision.

**Comment by ion-elgreco** [comment]
> > @stinodego I am addressing the overflow issue by casting to ms first in my other PR. If we merge that other one first, then I can update the test here, to reflect that change.
> 
> I don't think that's a valid way of doing things. You would lose a lot of precision on values tha require ns/us precision.

True but it's either using ms to prevent overflow in many cases, even though there is still a chance. Or we should just change the return type to float for duration variance.

**Comment by stinodego** [comment]
Closing in favor of #14493

**Pull request closed without merging**